Alternative implementation using Arrays.toString()

marcphilipp opened the pull request:
@sbrannen: Here's an alternative implementation that I find clearer and some additional tests. Please have a look and tell me what you think.

In addition, it does not require boxing of primitives.

ttddyy commented:
Also, I think factory loading mechanism in spring (e.g. loading `TestExecutionListener` from `spring.factory`) would be a good extension point for third party libraries.
Bootstrapping mechanism(`BootstrapContext`) in Spring TCF, as well.

sbrannen commented:
@ttddyy, wow... you discovered my _private_ playground and even opened an issue. :-o

Don't worry: we are definitely planning on ways for extensions to be auto-detected.

`TestDescriptor` is an SPI that will be implemented by what we are currently calling an `Engine`. Each engine will implement its own `TestDescriptor`. Furthermore, a `TestDescriptor` is not directly analogous to the `TestContext` in `spring-test`. For JUnit 5 we are planning something like a `TestExecutionContext` for that purpose.

For further comments, please see the junit-lambda project.

ttddyy commented:
oh, haha. it came up on my rss feed.

I have updated _["Extending JUnit's Standard Behaviour"](https://github.com/junit-team/junit-lambda/wiki/Extending-JUnit's-Standard-Behaviour#extension-points-to-replace-runners)_  wiki page for feature auto-detection as an extension point.

Latest junit-lambda code is looking great. :thumbsup:

thanks,

marcphilipp committed:
Alternative implementation using Arrays.toString()

------------------------------------------------------------------------
On behalf of the community, the JUnit Lambda Team thanks
AdNovum Informatik AG (http://www.adnovum.ch) for supporting the
JUnit crowdfunding campaign!
------------------------------------------------------------------------

marcphilipp committed:
Change tests to actually use primitive arrays

------------------------------------------------------------------------
On behalf of the community, the JUnit Lambda Team thanks
AdNovum Informatik AG (http://www.adnovum.ch) for supporting the
JUnit crowdfunding campaign!
------------------------------------------------------------------------
